### PR TITLE
Use conda-forge netcdftime wherever netcdf4 was tested 

### PR DESCRIPTION
**Pull request by jhamman**
- [x] Closes #1920
 - [x] Tests added (for all bug fixes or enhancements)
 - [x] Tests passed (for all non-documentation changes)
 - [x] Fully documented: see #1920

### CODE REVIEW

**Comment by fmaussion** [comment]
Is it safe to do so? I.e. can it hide bugs in the current netCDF4 time handling?

Also I think after #1920 it would be good to have a bit more info in the documentation about why people should switch to netcdftime or not.

**Comment by jhamman** [comment]
@fmaussion -

> Is it safe to do so? I.e. can it hide bugs in the current netCDF4 time handling?

Save the one test failure here, I think this is going to be a fairly safe change.  The `netcdftime` module in `netCDF4` is being ported to a stand-alone package (see: #1048, #1920, https://github.com/Unidata/netcdf4-python/pull/756, and #https://github.com/Unidata/netcdftime/issues/20).  The same test suite is being run on the `netcdftime` package and we've been working through a series of integration tests with xarray (mostly painless, i.e. #1929). 

> Also I think after #1920 it would be good to have a bit more info in the documentation about why people should switch to netcdftime or not.

Agreed. Do you think I should include that here?

**Comment by jhamman** [comment]
@fmaussion - mind reviewing the updated documentation here?

File: doc/installing.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ For netCDF and IO
 - `pynio <https://www.pyngl.ucar.edu/Nio.shtml>`__: for reading GRIB and other
   geoscience specific file formats
 - `zarr <http://zarr.readthedocs.io/>`__: for chunked, compressed, N-dimensional arrays.
-- `netcdftime <https://github.com/Unidata/netcdftime>`__: recommended if you
+- `netcdftime <https://unidata.github.io/netcdftime>`__: recommended if you
```
Comment (fmaussion) [inline doc/installing.rst]: add: ``or dates before year 1678 or after year 2262``

File: doc/time-series.rst [hunk 2]
```diff
@@ -47,6 +47,17 @@ attribute like ``'days since 2000-01-01'``).
 
 .. _CF conventions: http://cfconventions.org
 
+.. note::
+
+   When decoding/encoding datetimes for non-standard calendars or for dates
+   before year 1678 or after year 2262, xarray uses the `netcdftime`_ library.
+   ``netcdftime`` was previously packaged with the ``netcdf4-python`` package but
+   is now diestributed separately. ``netcdftime`` is an
```
Comment (spencerkclark) [inline doc/time-series.rst]: typo: distributed

**Comment by jhamman** [comment]
@shoyer - do you think this is good to merge in its current state?

**Comment by shoyer** [comment]
My understanding is that netcdftime hasn't had an official release yet?

Normally, you need to use some sort of prerelease channel and/or flag to download a package that hasn't had a normal release.

**Comment by jhamman** [comment]
right now, `conda install -c conda-forge netcdftime` will give you:

```
netcdftime:   1.0.0a2-py36_0      conda-forge
```

I don't andticipate any code changes before the final release. We're still working out the documentation site and whatnot.

`netcdf4` did just merge the removal of `netcdftime` (https://github.com/Unidata/netcdf4-python/pull/756) so this will need to go in prior to any release of netCDF4.

**Comment by shoyer** [comment]
Presumably netcdf4 will add a dependency on netcdftime so they don't break their API for users?

Anyways, if this already work then yes I'm good with merging this.

**Pull request merged**